app/lib/brexit_checker/account_jwt.rb

```diff
@@ -0,0 +1,33 @@
+class BrexitChecker::AccountJwt
+  def initialize(criteria_keys)
+    @criteria_keys = criteria_keys
+  end
+
+  def encode(key = ecdsa_key, algorithmn = "ES256")
+    JWT.encode payload, key, algorithmn
+  end
+
+private
+
+  attr_reader :criteria_keys
+
+  def payload
+    {
+      uid: uid,
+      data: criteria_keys,
+      scopes: scopes,
+    }
+  end
+
+  def uid
+    ENV.fetch("GOVUK_ACCOUNT_OAUTH_CLIENT_ID")
+  end
+
+  def scopes
+    "update-transition-checker-results"
+  end
+
+  def ecdsa_key
+    OpenSSL::PKey::EC.new(ENV.fetch("GOVUK_ACCOUNT_OAUTH_CLIENT_KEY"))
```

Comment:
This is a bit of a pain, means you need one for testing to work... open to better ideas?
Could stub ENV i guess in test... but i recall that having some wierdness associated that means everyone uses a gem.

Reply:
The scope is `transition_checker` and the attribute is `transition_checker_state`: https://github.com/alphagov/govuk-attribute-service-prototype/blob/main/config/scopes.yml#L8

Reply:
Yes, this will need to be stubbed, otherwise it will be impossible for developers to run the tests locally for the accounts flow.

I'd suggest adding a unit test for this class that stubs the environment variable.  Then for any tests that utilise this class, I'd stub the class.

Reply:
Cool thanks, this get stubbed [here](https://github.com/alphagov/finder-frontend/pull/2207/commits/12bb059c480c975e01d3f01d8722e8a3dbe216e0#diff-9b7921393a1b3bd053012ff427df2f90R3), a line which now sees my inbox being deluged with git guardian and "You've committed a secret!" spam ¬_¬